galaxy/tools/deps/container_resolvers/mulled.py

```diff
@@ -47,7 +50,13 @@ def output_line_to_image(line):
 
         if package_description.startswith("mulled-v1-"):
             hash = package_description
-            image = CachedMulledImageMultiTarget(hash, identifier)
+            image = CachedV1MulledImageMultiTarget(hash, identifier)
+        elif package_description.startswith("mulled-v2-"):
+            if version and "-" in version:
+                version_hash, build = version.split("-", 1)
+            else:
+                version_hash = None
```

Comment:
Yup - I hadn't tested that no versions case - I'll make another pass at this and fix that.